zap/src/main/java/org/zaproxy/zap/spider/parser/SpiderResourceFound.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2020 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.spider.parser;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import org.parosproxy.paros.network.HttpHeaderField;
+import org.parosproxy.paros.network.HttpMessage;
+import org.parosproxy.paros.network.HttpRequestHeader;
+
+/**
+ * Class SpiderResourceFound is used to store information about found resources by spider parsers.
+ *
+ * @since 2.11.0
+ */
+public class SpiderResourceFound {
```

Comment:
Better provide a builder, can be immutable.

Reply:
Resolved via last commit, SpiderResourceFound now offers a builder returning immutable instances